diff --git a/extension/parquet/include/writer/parquet_write_stats.hpp b/extension/parquet/include/writer/parquet_write_stats.hpp
index 1016c81fef..f6dea86b2a 100644
--- a/extension/parquet/include/writer/parquet_write_stats.hpp
+++ b/extension/parquet/include/writer/parquet_write_stats.hpp
@@ -10,6 +10,7 @@
 
 #include "column_writer.hpp"
 #include "geo_parquet.hpp"
+#include <limits>
 
 namespace duckdb {
 
@@ -77,6 +78,14 @@ class FloatingPointStatisticsState : public NumericStatisticsState<SRC, T, OP> {
 public:
 	bool has_nan = false;
 
+	FloatingPointStatisticsState() {
+		// initialize floating point stats to +/- infinity so that updates with
+		// -inf/+inf values are handled correctly (instead of using finite
+		// numeric limits which would cause incorrect max/min when the only
+		// value is infinite)
+		this->min = std::numeric_limits<T>::infinity();
+		this->max = -std::numeric_limits<T>::infinity();
+	}
 public:
 	bool CanHaveNaN() override {
 		return true;
diff --git a/src/storage/statistics/numeric_stats.cpp b/src/storage/statistics/numeric_stats.cpp
index 803c21f124..5391d625cf 100644
--- a/src/storage/statistics/numeric_stats.cpp
+++ b/src/storage/statistics/numeric_stats.cpp
@@ -19,8 +19,16 @@ BaseStatistics NumericStats::CreateUnknown(LogicalType type) {
 BaseStatistics NumericStats::CreateEmpty(LogicalType type) {
 	BaseStatistics result(std::move(type));
 	result.InitializeEmpty();
-	SetMin(result, Value::MaximumValue(result.GetType()));
-	SetMax(result, Value::MinimumValue(result.GetType()));
+
+	if (result.GetType() == LogicalType::FLOAT || result.GetType() == LogicalType::DOUBLE) {
+		SetMin(result, Value::Infinity(result.GetType()));
+		SetMax(result, Value::NegativeInfinity(result.GetType()));
+	}
+	else {
+		SetMin(result, Value::MaximumValue(result.GetType()));
+		SetMax(result, Value::MinimumValue(result.GetType()));
+	}
+	
 	return result;
 }
 
