diff --git a/src/storage/postgres_table_set.cpp b/src/storage/postgres_table_set.cpp
index 9f5e7fc..1456608 100644
--- a/src/storage/postgres_table_set.cpp
+++ b/src/storage/postgres_table_set.cpp
@@ -27,7 +27,8 @@ string PostgresTableSet::GetInitializeQuery(const string &schema, const string &
 
 string PostgresTableSet::GetInitializeQuery(const vector<string> &schemas, const string &table) {
 	string base_query = R"(
-SELECT pg_namespace.oid AS namespace_id, relname, relpages, attname,
+SELECT pg_namespace.oid AS namespace_id, relname,
+    pg_relation_size(pg_class.oid) / current_setting('block_size')::bigint AS relpages, attname,
     pg_type.typname type_name, atttypmod type_modifier, pg_attribute.attndims ndim,
     attnum, pg_attribute.attnotnull AS notnull, NULL constraint_id,
     NULL constraint_type, NULL constraint_key, type_ns.nspname AS type_schema,
